utilities_common/routing_stack.py

```diff
@@ -0,0 +1,116 @@
+import subprocess
+from utilities_common import constants
+
+"""
+Get the current routing stack name
+:param extended_validation: Whether to run the an extended routing stack check to provide the
+                            user with info about the routing stack state when running show
+                            commands. If routing stack state is stopped or restarting,
+                            some show commands will be unavailable
+:return:                    The routing-stack name, if present
+                            The routing-stack state
+"""
+def get_routing_stack(extended_validation=False):
+    routing_stack_state = constants.ROUTING_STACK_STATE_RUNNING
+    if extended_validation:
+        ROUTING_STACK_FAILED = "\nRouting-stack has exited with an error\n" \
+                               "Show commands dependent on routing stack are not available"
+        ROUTING_STACK_RESTARTING = "\nRouting stack currently starting/restarting. \n " \
+                                   "Expect routing-stack related show commands to be unrecognized for a while...\n"
+
+        # First check for the routing-stack name among the failed containers
+        chk_status_cmd = "sudo docker ps -f status=removing -f status=exited -f status=dead | grep bgp | awk '{print$8}'"
+
+        try:
+            proc = subprocess.Popen(chk_status_cmd,
+                                    stdout=subprocess.PIPE,
+                                    shell=True,
+                                    text=True)
+            stdout = proc.communicate()[0]
+            proc.wait()
+            chk_status_result = stdout.rstrip('\n')
+
+        except OSError as e:
+            raise OSError("Cannot get status from routing stack")
```

Comment:
>OSError [](http://example.com/codeflow?start=18&length=7)

You raise the same exception type but loss all its old context, such as stack, details. Could you improve?